task/README.md

```diff
@@ -0,0 +1,272 @@
+| `maxgas`  | Integer               | Maximum gas for the invocation            | No       | 1000 <!-- ...or something --> |
+
+## 3.1 `type`
+
+The `type` field declares this object to be an IPVM Wasm configuration. The value MUST be `ipvm/wasm`.
+
+## 3.2 `with`
+
+The `with` field declares the Wasm module to load via CID.
+
+Note that the 
+
+## 3.3 `input`
+
+## 3.4 `maxgas`
+
+The `maxgas` field specifies the upper limit in gas that this task may consume.
+
+For the gas schedule, please see the [gas schedule spec].
+
+# 4 Effects
+
+The contract for effects is different from pure computation. As effects by definition interact with the "real world". These may be either commands or queries. Exmaples of effects include reading from DNS, sending an HTTP POST request, running a WASI module with network access, or receieving a random value.
+
+The `with` field MAY be filled from a relative value (previous step)
+
+| Field     | Type            | Description                         | Required | Default |
+|-----------|-----------------|-------------------------------------|----------|---------|
+| `type`    | `"ipvm/effect"` | Identify this workflow as an effect | Yes      |         |
+| `version` | SemVer          | IPVM effect schema version          | No       | `0.1.0` |
+| `using`   | URI             |                                     | Yes      |         |
+| `do`      | ability         |                                     | Yes      |         |
+| `args`    | `[{}]`          |                                     | No       | `[]`    |
+| `timeout` | Integer         | Timeout in milliseconds             | No       | `5000`  |
+| `auth`    | `[&UCAN]`       |                                     | No       | `[]`    |
+
+<!-- | `destructive` | Boolean         |  FIXME infer from `do` field?  | No       | `True`  | -->
+<!-- indeed these need to be registered by the runner -->
+
+``` json
+{
+    "type": "ipvm/effect",
+    "to": "did:key:zStEZpzSMtTt9k2vszgvCwF4fLQQSyA15W5AQ4z3AR6Bx4eFJ5crJFbuGxKmbma4",
+    "do": "crypto/sign",
+    "args": [
+        { "value": { "from": "earlierStep" } }
+    ]
+}
+```
+
+## 4.1 `type`
+
+
+``` json
+{
+  "type": "ipvm/pure",
+  "version": "0.1.0",
+  "using": "wasm:Qm12345"
+  "args": {
+    "type": "ipvm/wasm",
+    "version": "0.1.0",
+    "fun": "add_one",
+    "args": [1, 2, 3],
+    "maxgas": 1024
+  }
+}
+```
+
+``` json
+{
+  "type": "ipvm/effect",
+  "version": "0.1.0",
+  "using": "docker:Qm12345"
+  "meta": {
+    "description": "Tensorflow container",
+    "tags": ["machine-learning", "tensorflow", "myproject"]
+  },
+  "do": {
+    "resources": {
+      "ram": {"gb": 10}
+    },
+    "inputs": [1, 2, 3],
+    "entry": "/",
+    "workdir": "/",
+    "env": {
+      "$FOO": "bar"
+    },
+    "timeout": {"seconds": "3600"},
+    "contexts": [],
+    "output": [],
+    "sharding": 5
+  }
+}
+```
```

Comment:
I'm not sure I've really understood the intention around effects w.r.t. the rest of the objects in this spec... it seems like an effect IS a task? But I also feel like these things are very different? E.g. the effect you have on line 215 looks very different from this one. 

Is an effect really a wrapper around a task? So an effect may be "invoke another task" but it may also be something else?